### ISSUE TITLE
Need better linting for "Snapshots in callbacks are not recommended" when importing another file

### ISSUE THREAD

**Issue by devnomic** [pmndrs/eslint-plugin-valtio#39]
Lets say i have a hook file with a state in it. `useAppState.ts`

````
export const state = proxy({
  isLocked: false,
});

export const useAppState = () => {
  const snap = useSnapshot(state);
  return {
    ...snap,
    state,
  };
};

````

File 2 `SomeComponent.tsx`

````
const { state, isLocked } = useAppState();

const handleClick = () => {
   console.log(isLocked); // No warning
};

````

Hopefully this can be resolved also, as this will probably lead to hard to catch bugs.

**Comment by dai-shi** [comment]
Hi, thanks for opening this up.

```js
  return {
    ...snap,
    state,
  };
```
That seems like almost impossible to track.

```js
const { state, isLocked } = useAppState();

const handleClick = () => {
   console.log(isLocked); // No warning
};
```
It looks actually fine if `isLocked` isn't an object.

> Need better linting for "Snapshots in callbacks are not recommended" when importing another file

Oh, do you mean, it's a false positive?

**Comment by devnomic** [comment]
Hi @dai-shi 

Thanks for replying, no it was not false positive.
It was detected just fine by eslint, if the state is not imported.
don't know why?

e.g.

````
const state = proxy({
  isLocked: false,
});

function SomeComponent() {
   const { isLocked } = useSnapshot(state);
   const handleClick = () => {
     console.log(isLocked); // Correct, lint warn
  };
}

````

**Comment by devnomic** [comment]
I think maybe not detected because of spreading

**Comment by dai-shi** [comment]
I see. (I misunderstood something in my previous comment.)

Yeah, maybe. Can you try without spreading?
